src/ConnectionsListener.cpp

```
@@ -79,6 +80,8 @@ void ConnectionsListener::clearClientFd(int clientId) {
 void ConnectionsListener::requestExit() {
     Mutex::Autolock al(mMutex);
     mRunning = false;
+    for (int clientId = 0; clientId < mClientFdPromises.size(); clientId++)
+        mClientFdPromises[clientId].set_value(-1);
```

You might have to check for mClientsConnected[clientId] before setting this value.
Because, if you try to set the value for a promise which has already been set earlier, it might cause an exception.

Does this scenario work fine in gtest : 
Start vhal > Connect Streamer > Attempt cleanup(requestExit).
By the time you call requestExit, mClientFdPromises[0] already has a value set from the streamer connection. So, it should throw an exception.